fe/src/main/java/org/apache/doris/analysis/CreateMaterializedViewStmt.java

```diff
@@ -255,10 +259,15 @@ private void analyzeOrderByClause() throws AnalysisException {
                 MVColumnItem mvColumnItem = mvColumnItemList.get(i);
                 Expr resultColumn = selectStmt.getResultExprs().get(i);
                 keyStorageLayoutBytes += resultColumn.getType().getStorageLayoutBytes();
-                if ((i + 1) <= FeConstants.shortkey_max_column_count
-                        || keyStorageLayoutBytes < FeConstants.shortkey_maxsize_bytes) {
+                if ((!mvColumnItem.getType().isFloatingPointType())
```

Comment:
We should be break if we met a floating point type column. The following case will get wrong result in your logic:

`k1 int, k2 float, k3 int`

Reply:
Why?